source/MRMesh/MRPointsComponents.cpp

```diff
@@ -0,0 +1,157 @@
+#include "MRPointsComponents.h"
+#include "MRPointCloud.h"
+#include "MRBitSet.h"
+#include "MRTimer.h"
+#include "MRPointsInBall.h"
+#include "MRProgressCallback.h"
+#include "MRPch/MRTBB.h"
+
+namespace MR
+{
+
+namespace PointCloudComponents
+{
+
+Expected<MR::VertBitSet> getLargestComponentsUnion( const PointCloud& pointCloud, float maxDist, int minSize, ProgressCallback pc /*= {}*/ )
+{
+    MR_TIMER
+
+    assert( maxDist > 0.f );
+    assert( minSize > 0 );
+    const auto& validPoints = pointCloud.validPoints;
+    ProgressCallback subPc = pc ? subprogress( pc, 0.f, 0.9f ) : pc;
```

Comment:
no need in `pc ?`. This check is already in `subprogress`

Reply:
fixed